src/Main.elm

```diff
@@ -1,16 +1,219 @@
+import Css.Foreign exposing (body, global)
+import Css.Helpers exposing (toCssIdentifier)
+import Data.Audience exposing (audiencesDecoder, audiencesJSON)
+import Data.AudienceFolder exposing (audienceFoldersDecoder, audienceFoldersJSON)
+import Dict
+import FontAwesome exposing (edit, folder, folderOpen, icon, useCss)
+import Html
+import Html.Styled exposing (Html, button, div, fromUnstyled, li, styled, text, toUnstyled, ul)
+import Html.Styled.Attributes exposing (class, classList)
+import Html.Styled.Events exposing (onClick)
+import Json.Decode exposing (decodeString)
+import NaturalOrdering
 
 
--- Import Modules
+bodyStyleNode : Html msg
+bodyStyleNode =
+    global
+        [ body
+            [ backgroundColor (hex "#191e29")
+            , color (hex "#b3c4cb")
+            , fontFamilies [ "sans-serif" ]
+            , boxSizing borderBox
+            ]
+        ]
 
-import Data.Audience
-import Data.AudienceFolder
 
+type ContentListClasses
+    = FolderClass
+    | FolderOpenedClass
+    | ContentListClass
+    | ContentElementClass
+    | ContentElementTextClass
+    | ContentElementIconClass
+    | FolderCountClass
+    | AudienceEditIconClass
 
-{-| Main file of application
--}
-main : Html msg
+
+listStyleNode : Html msg
+listStyleNode =
+    Css.Foreign.global
+        [ Css.Foreign.class ContentListClass
+            [ listStyleType none
+            , padding2 (px 2) zero
+            , backgroundColor (hex "#fff")
+            , width (px 300)
+            , maxHeight (px 600)
+            , overflowY auto
+            ]
+        , Css.Foreign.class ContentElementClass
+            [ borderRadius (px 4), margin (px 6), padding (px 10), minHeight (px 40), displayFlex, alignItems center, fontSize (px 13), color (hex "#e2f1f8"), backgroundColor (hex "#4294ca") ]
+        , Css.Foreign.class ContentElementTextClass
+            [ textOverflow ellipsis, overflowX hidden ]
+        , Css.Foreign.class ContentElementIconClass
+            [ minWidth (px 40), displayFlex, justifyContent center ]
+        , Css.Foreign.class FolderClass
+            [ cursor pointer ]
+        , Css.Foreign.class FolderOpenedClass
+            [ color (hex "#ecf2f6"), backgroundColor (hex "#346f97"), cursor default ]
+        , Css.Foreign.class FolderCountClass
+            [ borderRadius (px 4), color (hex "#779aac"), backgroundColor (hex "#295a78"), padding (px 4), marginLeft (px 6) ]
+        , Css.Foreign.class AudienceEditIconClass
+            [ marginLeft auto, minWidth (px 30), displayFlex, justifyContent flexEnd ]
+        ]
+
+
+type alias Model =
+    { breadcrumbs : List Int
```

Comment:
do you really need list in here? I think you can just store `Maybe Int` of the current folder. Ancestors are then recursive function. There would some performance penalty for those lookups but it will limit state this component manages so less potential bugs and state synchronizations when structure starts to be editable.